src/Transfer/Worker.cpp

```diff
@@ -212,21 +212,124 @@ Worker::processCollisions(
   }
 
   for (int i = 0; i < numchares; i++) {
-    proxy[i].checkPoints(separated[i].size(), separated[i].data());
+    proxy[i].determineActualCollisions(separated[i].size(), separated[i].data());
   }
   delete[] separated;
 }
 
 void
-Worker::checkPoints(
+Worker::determineActualCollisions(
     int nPoints,
     std::pair<CkVector3d, int>* points )
 // *****************************************************************************
-//
+//  Identify actual collisions by calling intet function on all possible collisions
+//! \param[in] nPoints Number of points to be checked
+//! \param[in] points Pairs of point and tet indices
 // *****************************************************************************
 {
-  CkPrintf("[%i]: Received data for %i potential collisions with my tets\n",
-      CkMyPe(), nPoints);
+  CkPrintf("[%i]: Received data for %i potential collisions with my tets\n", CkMyPe(), nPoints);
+
+  std::array< real, 4 > N;
+  int numInTet = 0;
+
+  // Iterate over my potential collisions and determine if it is intet and the shapefunction
+  for(int i = 0; i < nPoints; i++) {
+    bool val = intet(points[i].first,
+                     points[i].second,
+                     N);
+
+    if(val)
+      numInTet++;
+
+    // TODO: I'm guessing we need the point(points[i]), the tet index(points[i].second), and the shapefunction(N) to be stored from this function for future use?
+  }
```

Comment:
Yes, we will use the finite element shape functions, evaluated at the point location, to interpolate the results between meshes. (The points/vertices of the destination mesh will fall into tets of the source mesh whose vertices will hold the current solution to be interpolated.)